### ISSUE TITLE
[$250] Hide current user in approver selector when 'prevent self approvals' is enabled

### ISSUE THREAD

**Issue by marcaaron** [Expensify/App#53612]
cc @JmillsExpensify coming from our DM

While we have the ability to set "Prevent self approval" on a workspace. There are some ways in which this feature works unexpectedly in NewDot. Specifically, the following scenarios seem possible and should not be:

## Incorrect workflow setup restrictions

**Problem**

- User tries to create a new workflow with themselves as approver when prevent self approval is enabled.
- This should not be possible. The OldDot behavior is to show a "problem" that needs to be fixed like this:

![image](https://github.com/user-attachments/assets/4602073f-ed05-4f7b-a0fc-9105ebe2b1af)

In NewDot, we just let it happen:

![Screenshot 2024-12-04 at 1 24 36 PM](https://github.com/user-attachments/assets/e10af23b-dec3-48ec-a59b-14c15f47fd4b)
![Screenshot 2024-12-04 at 1 24 58 PM](https://github.com/user-attachments/assets/20edde92-59a9-445f-8317-9622d1f7c36f)

**Solution:**

- Ideally, we would just remove this person from the list of approvers when giving the option to select an approver

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021866263734682899164</li>
        <li>Upwork Job ID: 1866263734682899164</li>
        <li>Last Price Increase: 2024-12-09</li>
<li>Automatic offers: </li>
<ul>
<li>brunovjk | Reviewer | 105298681</li>
<li>Krishna2323 | Contributor | 105298682</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @brunovjk</details>

**Comment by marcaaron** [comment]
I think this part: `Incorrect workflow setup restrictions` can be external as it's straightforward enough. 

How to handle the users who are already self-approvers is less clear to me and probably requires a non-trivial backend change and maybe better to discuss in #product first. I feel like this is an edge case related to Workflows feature (so cc'ing @tgolen in case he has thoughts on this one as well).

**Comment by marcaaron** [comment]
@Krishna2323 looks good, but not ready for proposals just yet.

**Comment by marcaaron** [comment]
Ah, so, maybe it is a bit easier to have them set back to whatever the default approver here is? 

![Screenshot 2024-12-05 at 11 45 52 AM](https://github.com/user-attachments/assets/60a4af1e-3bdb-4888-9da4-057397281780)

Or are we wanting to update each custom workflow to the workspace owner with some kind of note?

Either way, I think we may need a backend change since all the affected reports would need to have their `managerID` set to a new approver (and potentially Onyx updates sent out for this change).

>potentially highlight the specific workflows that have changed (and if there are several, then we make sure at least one is within the scroll view and is highlighted

The user updating the prevent self approval setting would be on the "Rules" page. I think it would be easier to update them without having any kind of highlighting, but I might be missing why that's important.

**Comment by JmillsExpensify** [comment]
Yeah great call! Let's set them back to the default approver. In that case, also no need for highlighting.

**Comment by JmillsExpensify** [comment]
@marcaaron do you think we're ready to open up to proposals following our discussion last week?

**Comment by marcaaron** [comment]
Yeah. Ok, I think for clarity I am going to split this into two separate issues and we will make this one External and only focus on the exclusion of the user from the person selector page.

>@JmillsExpensify, do we want to show the modal or we just want to exclude the members from the approvers list?

Let's chat about this in the next issue.

**Comment by brunovjk** [comment]
> only focus on the exclusion of the user from the person selector page

With that I think we can go with @Krishna2323's [proposal](https://github.com/Expensify/App/issues/53612#issuecomment-2518852101)

 🎀👀🎀 C+ reviewed

**Comment by brunovjk** [comment]
I think the automation failed to update the status. The [PR](https://github.com/Expensify/App/pull/54005) was deployed to production on [December 18th](https://github.com/Expensify/App/issues/54111#issuecomment-2551261636).

**Comment by bfitzexpensify** [comment]
Paid out @brunovjk - looks like we're still waiting on @Krishna2323 to accept the Upwork job

**Comment by brunovjk** [comment]
@marcaaron Do you think we should create a regression test here? Thanks.

**Comment by brunovjk** [comment]
# BugZero Checklist:

- [x] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [x] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [x] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [x] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [x] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment: N/A (No specific offending PR identified;

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [ ] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

## Regression Test Proposal

### Test:
1. Create a workspace > Add few members
2. Enable workflows & rules
3. Open workflows page > enable "Add approvals"
4. Open rules page > enabled "Prevent self-approvals"
5. Open workflows page > "Add approval workflow"
6. Select few members for "Expenses from" > Click "Next"
7. Verify selected members from previous steps does not appear on "Approver" page.
8. Select a approver and continue
9. Now select Expenses from field > verify the approver doesn't appear on Expenses from page

Do we agree 👍 or 👎

**Comment by brunovjk** [comment]
> Paid out @brunovjk - looks like we're still waiting on @Krishna2323 to accept the Upwork job

@Krishna2323 Did you get a chance to accept it?

**Comment by JmillsExpensify** [comment]
Both contributors paid out, and regression test created. Closing this issue.
